src/hooks/useSwapEventsSubscriber.ts

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState } from 'react';
+
+import { Swap } from '@airswap/libraries/build/src/Contracts';
+import { useWeb3React } from '@web3-react/core';
+import { BigNumber } from 'ethers';
+import { noop } from 'react-use/lib/misc/util';
+
+import { isEqualAddress } from '../helpers/string';
+import { useAppDispatch, useAppSelector } from '../redux/hooks';
+import {
+  fetchCurrencyTokenAllowance,
+  fetchCurrencyTokenBalance,
+  fetchUserTokens,
+} from '../redux/stores/balances/balancesApi';
+import { setTokens } from '../redux/stores/balances/balancesSlice';
+import { setOrders } from '../redux/stores/collection/collectionSlice';
+import { setOrders as setProfileOrders } from '../redux/stores/profileOrders/profileOrdersSlice';
+import { addNftSoldToast } from '../redux/stores/toasts/toastsActions';
+
+interface LastSoldOrder {
+  nonce: string;
+  tokenId: string;
+}
+
+const useSwapEventsSubscriber = () => {
+  const { account } = useAppSelector(state => state.web3);
+  const { chainId, collectionToken, currencyToken } = useAppSelector(state => state.config);
+  const { tokens } = useAppSelector(state => state.balances);
+  const { orders } = useAppSelector(state => state.collection);
+  const { orders: profileOrders } = useAppSelector(state => state.profileOrders);
+
+  const dispatch = useAppDispatch();
+  const { provider: library } = useWeb3React();
+
+  const [lastSoldOrder, setLastSoldOrder] = useState<LastSoldOrder>();
+
+  useEffect(() => {
+    if (
+      !lastSoldOrder
+        || !tokens.includes(lastSoldOrder.tokenId)
+        || !library
+        || !account
+    ) {
+      return;
+    }
+
+    const newOrders = orders.filter(order => order.nonce !== lastSoldOrder.nonce);
+    const newProfileOrders = profileOrders.filter(order => order.nonce !== lastSoldOrder.nonce);
+    const newUserTokens = tokens.filter(token => token !== lastSoldOrder.tokenId);
+
+    setLastSoldOrder(undefined);
+
+    dispatch(addNftSoldToast(lastSoldOrder.tokenId));
+    dispatch(setProfileOrders(newProfileOrders));
+    dispatch(setTokens(newUserTokens));
+    dispatch(setOrders(newOrders));
+
+    dispatch(fetchCurrencyTokenBalance({
+      chainId,
+      provider: library,
+      collectionTokenAddress: collectionToken,
+      walletAddress: account,
+    }));
+
+    dispatch(fetchCurrencyTokenAllowance({
+      chainId,
+      provider: library,
+      collectionTokenAddress: collectionToken,
+      walletAddress: account,
+    }));
+
+    dispatch(fetchUserTokens({
+      provider: library,
+      walletAddress: account,
+      collectionToken,
+    }));
+  }, [
+    lastSoldOrder,
+    tokens,
+    library,
+    account,
+  ]);
+
+  const onSwap = async (
+    nonce: BigNumber,
+    signerWallet: string,
+    signerAmount: BigNumber,
```

Comment:
This isn't problematic, but I see a few unused variables in this `onSwap` function: `signerAmount`, `senderWallet`, `senderAmount`, and `senderTokenId`

Reply:
Yes I can't omit those because I need the last `senderToken` parameter.